event-sourcing/event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/EventStreamManager.java

```diff
@@ -179,33 +171,41 @@ public UUID cloneAsAncestor(final UUID id) throws EventStreamException {
      * Clears the stream, deleting all associated events from the event_log, it does not update the
      * event_stream.
      */
-    public void clear(final UUID id) {
-        eventRepository.clear(id);
+    public void clearEventsForStream(final UUID id) {
+        eventRepository.clearEventsForStream(id);
     }
 
     /**
-     * Get the current (current maximum) sequence id (version number) for a stream
+     * Get the latest position number for a stream
      *
      * @param id the id of the stream
-     * @return the latest sequence id for the provided steam. 0 when stream is empty.
+     * @return the latest position number for the provided steam. 0 when stream is empty.
      */
-    public long getCurrentVersion(final UUID id) {
-        return eventRepository.getCurrentSequenceIdForStream(id);
+    public long getCurrentPosition(final UUID id) {
+        return eventRepository.getCurrentPositionForStream(id);
     }
 
-    private long append(final UUID id, final Stream<JsonEnvelope> events, final Optional<Long> versionFrom) throws EventStreamException {
+    /**
+     * Get the latest position number of the stream within the streams
+     ** @return the latest position number for the provided steam. 0 when stream is empty.
```

Comment:
This is wrong; it is return the position of the stream; it'll never be zero.

Reply:
Updated